libvast/src/system/meta_index.cpp

```diff
@@ -329,7 +333,12 @@ std::vector<uuid> meta_index_state::lookup_impl(const expression& expr) const {
         },
         [&](const field_extractor& lhs, const data&) -> result_type {
           auto pred = [&](const auto& field) {
-            return field.name().ends_with(lhs.field);
+            VAST_ASSERT(!field.is_standalone_type());
+            auto rt = record_type{{field.field_name(), field.type()}};
+            for ([[maybe_unused]] const auto& offset :
+                 rt.resolve_key_suffix(lhs.field, field.layout_name()))
+              return true;
```

Comment:
Can we do a type check here as well? That would help in case we have multiple fields with the same suffix name but different types.